icicle/backend/cpu/src/curve/cpu_msm.hpp

```diff
@@ -65,11 +86,13 @@ class EcAddTask : public TaskBase
    */
   void set_phase1_addition_with_affine(const P& bucket, const A& base, int bucket_idx)
   {
-    m_point1 = bucket;
-    m_point2_affine = base;
-    m_return_idx = bucket_idx;
-    m_point2_opcode = ADD_P1_AND_P2_AFFINE;
-    dispatch();
+    m_a_points[m_nof_valid_points] = bucket;
+    m_b_affine_points[m_nof_valid_points] = base;
+    m_return_idx[m_nof_valid_points] = bucket_idx;
+    m_opcodes[m_nof_valid_points] = ADD_P1_AND_P2_AFFINE;
+
+    m_nof_valid_points++;
+    if (m_nof_valid_points == EC_BATCH_SIZE) { dispatch(); }
```

Comment:
what if it doesn't divide? will not you dispatch?

Reply:
No, it will the MSM itself dispatches the last tasks (see line 498 for an example at the end of phase 1).
This check comes to save the MSM code itself from checking every time it adds another addition to the task (Which would be more messy in the code).